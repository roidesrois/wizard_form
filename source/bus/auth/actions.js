// Types
import { types } from './types';

export const authActions = {
    // Sync
    initialize: () => ({
        type: types.INITIALIZE,
    }),
    authenticate: () => ({
        type: types.AUTHENTICATE,
    }),
    logout: () => ({
        type: types.LOGOUT,
    }),
    valid: () => ({
        type: types.VALID,
    }),
    registered: () => ({
        type: types.REGISTERED,
    }),

    // Async
    initializeAsync: () => ({
        type: types.INITIALIZE_ASYNC,
    }),
    loginAsync: (credentials) => ({
        type:    types.LOGIN_ASYNC,
        payload: credentials,
    }),
    authenticateAsync: () => ({
        type: types.AUTHENTICATE_ASYNC,
    }),
    signupAsync: (userData) => ({
        type:    types.SIGNUP_ASYNC,
        payload: userData,
    }),
    confirmSignupAsync: (registrationHash) => ({
        type:    types.CONFIRM_SIGNUP_ASYNC,
        payload: registrationHash,
    }),
    activateAccountAsync: (userData) => ({
        type:    types.ACTIVATE_ACCOUNT_ASYNC,
        payload: userData,
    }),
    logoutAsync: () => ({
        type: types.LOGOUT_ASYNC,
    }),
};
