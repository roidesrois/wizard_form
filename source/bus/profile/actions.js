// Types
import { types } from './types';

export const profileActions = {
    // Sync
    fillProfile: (profile) => ({
        type:    types.FILL_PROFILE,
        payload: profile,
    }),
    updateAvatar: (newAvatarUrl) => ({
        type:    types.UPDATE_AVATAR,
        payload: newAvatarUrl,
    }),
    clearProfile: () => ({
        type: types.CLEAR_PROFILE,
    }),

    // Async
    updateNameAsync: (newName) => ({
        type:    types.UPDATE_NAME_ASYNC,
        payload: newName,
    }),
    updateAvatarAsync: (newAvatar) => ({
        type:    types.UPDATE_AVATAR_ASYNC,
        payload: newAvatar,
    }),
    updatePasswordAsync: (newPassword) => ({
        type:    types.UPDATE_PASSWORD_ASYNC,
        payload: newPassword,
    }),
    updateEmailAsync: (newEmail) => ({
        type:    types.UPDATE_EMAIL_ASYNC,
        payload: newEmail,
    }),
    resetPasswordAsync: (email) => ({
        type:    types.RESET_PASSWORD_ASYNC,
        payload: email,
    }),
};
