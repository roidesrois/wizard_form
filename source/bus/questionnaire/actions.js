// Types
import { types } from './types';

export const questionnaireActions = {
    // Sync
    fillQuestionnaire: (questionnaire) => ({
        type:    types.FILL_QUESTIONNAIRE,
        payload: questionnaire,
    }),

    // setQuestionnaireValidation: (validation) => ({
    //     type:    types.SET_QUESTIONNAIRE,
    //     payload: validation,
    // }),

    // Async
    fetchQuestionnaireAsync: () => ({
        type: types.FETCH_QUESTIONNAIRE_ASYNC,
    }),
};
