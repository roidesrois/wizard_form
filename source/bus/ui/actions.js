// Core
import { v4 } from 'uuid';

// Instruments
import { types } from './types';

export const uiActions = {
    startFetching: () => ({
        type: types.START_FETCHING,
    }),
    stopFetching: () => ({
        type: types.STOP_FETCHING,
    }),
    emitError: (error, meta = null) => ({
        type:    types.EMIT_ERROR,
        payload: error,
        error:   true,
        meta,
    }),
    showNotification: (message, type = 'info', source = '') => ({
        type:    types.SHOW_NOTIFICATION,
        payload: {
            id: v4(),
            message,
            type,
            source,
        },
    }),
    hideNotification: () => ({
        type: types.HIDE_NOTIFICATION,
    }),
    setOnlineState: () => ({
        type: types.SET_ONLINE_STATE,
    }),
    setOfflineState: () => ({
        type: types.SET_OFFLINE_STATE,
    }),
};
