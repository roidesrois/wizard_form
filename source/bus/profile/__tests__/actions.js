// Core
import { profileActions } from '../actions';

const {
    testData: { userProfile, url },
} = global;

describe('profile actions:', () => {
    // Sync
    test('fillProfile snapshot', () => {
        expect(profileActions.fillProfile(userProfile)).toMatchSnapshot();
    });

    test('updateAvatar snapshot', () => {
        expect(profileActions.updateAvatar(url)).toMatchSnapshot();
    });

    test('clearProfile snapshot', () => {
        expect(profileActions.clearProfile()).toMatchSnapshot();
    });

    // Async
    test('updateNameAsync snapshot', () => {
        expect(profileActions.updateNameAsync(userProfile)).toMatchSnapshot();
    });

    test('updateAvatarAsync snapshot', () => {
        expect(profileActions.updateAvatarAsync(userProfile)).toMatchSnapshot();
    });

    test('updatePasswordAsync snapshot', () => {
        expect(
            profileActions.updatePasswordAsync(userProfile),
        ).toMatchSnapshot();
    });
});
