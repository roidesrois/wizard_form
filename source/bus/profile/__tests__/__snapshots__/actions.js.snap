// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile actions: clearProfile snapshot 1`] = `
Object {
  "type": "CLEAR_PROFILE",
}
`;

exports[`profile actions: fillProfile snapshot 1`] = `
Object {
  "payload": Object {
    "avatar": "TEST_AVATAR",
    "firstName": "TEST_FIRST_NAME",
    "id": "TEST_ID",
    "lastName": "TEST_LAST_NAME",
    "token": "TEST_TOKEN",
  },
  "type": "FILL_PROFILE",
}
`;

exports[`profile actions: updateAvatar snapshot 1`] = `
Object {
  "payload": "https://www.url.com",
  "type": "UPDATE_AVATAR",
}
`;

exports[`profile actions: updateAvatarAsync snapshot 1`] = `
Object {
  "payload": Object {
    "avatar": "TEST_AVATAR",
    "firstName": "TEST_FIRST_NAME",
    "id": "TEST_ID",
    "lastName": "TEST_LAST_NAME",
    "token": "TEST_TOKEN",
  },
  "type": "UPDATE_AVATAR_ASYNC",
}
`;

exports[`profile actions: updateNameAsync snapshot 1`] = `
Object {
  "payload": Object {
    "avatar": "TEST_AVATAR",
    "firstName": "TEST_FIRST_NAME",
    "id": "TEST_ID",
    "lastName": "TEST_LAST_NAME",
    "token": "TEST_TOKEN",
  },
  "type": "UPDATE_NAME_ASYNC",
}
`;

exports[`profile actions: updatePasswordAsync snapshot 1`] = `
Object {
  "payload": Object {
    "avatar": "TEST_AVATAR",
    "firstName": "TEST_FIRST_NAME",
    "id": "TEST_ID",
    "lastName": "TEST_LAST_NAME",
    "token": "TEST_TOKEN",
  },
  "type": "UPDATE_PASSWORD_ASYNC",
}
`;
